client/lib/reader-post-flux-adapter/index.js

```diff
@@ -42,6 +42,16 @@ const fluxPostAdapter = Component => {
 		state = this.getStateFromStores( this.props );
 
 		updateState = ( newState = this.getStateFromStores() ) => {
+			// check to see if this new state is the same as the old state
+			if ( newState.posts.length === this.state.posts.length ) {
```

Comment:
and yes, it would still avoid the render if we had it in shouldComponentUpdate. But this avoids needing that at all.